Mathlib/Geometry/Manifold/Notation.lean

```diff
@@ -178,6 +180,25 @@ scoped elab:max "T% " t:term:arg : term => do
         mkLambdaFVars #[x] body
   | _ => return e
 
+open Elab in
+/--
+Elaborator for sections in a fibre bundle: converts a section `s : Π x : M, V x` as a dependent
+function to a non-dependent function into the total space. This handles the cases of
+- sections of a trivial bundle
+- vector fields on a manifold (i.e., sections of the tangent bundle)
+- sections of an explicit fibre bundle
+- turning a bare function `E → E'` into a section of the trivial bundle `Bundle.Trivial E E'`
+
+This elaborator searches the local context for suitable hypotheses for the above cases by matching
+on the expression structure, avoiding `isDefEq`. Therefore, it is (hopefully) fast enough to always
+run.
+-/
+scoped elab:max "T% " t:term:arg args:term* : term => do
+  let e ← Term.elabTerm t none
+  let args' ← args.mapM (fun t ↦ Term.elabTerm t none)
+  trace[Elab.DiffGeo.TotalSpaceMk] "argument(s) passed to `T%` is/are `{args}`"
+  totalSpaceElab (Expr.beta (Expr.eta e) args')
```

Comment:
Huh, I'm really surprised if this affects anything at the type level! (Not saying it necessarily doesn't, though.) This having an effect means that the *type* of `mkAppN e args'` depends on whether `mkAppN e args'` itself is head-eta and beta reduced! Is that possible?

And/or, what's an example of one of the weird type errors we're seeing, or the kind of issue we hit (even if we don't have a test for the fix yet)?

And just to be clear--the primary point of this PR is to change the `#check` and hover experience when using `T%`, right? Any type mismatch fixing would just be a bonus? (Sorry for the multiple edits to this comment here!)